app/controllers/application_controller.rb

```diff
@@ -39,6 +40,7 @@ class ApplicationController < Sinatra::Base
   end
 
   get '/sessions/logout' do
+    session.clear
```

Comment:
Also upon log out (the '/sessions/logout' controller action block), we need to clear the session of all of it's content before redirecting back to the home page.